### PR TITLE
fix(Card)!: Animation Fixes and Gamut migration

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

- Fixes animation bug where background bleed through would happen erroneously (Pseudo element getting clipped).
- Migrates component to gamut
- Cleans up some of the logic and brings it inline with our current practices.

Monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/22897

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.14-alpha.81eea5.0<br/>@codecademy/gamut-labs@11.2.2-alpha.81eea5.0<br/>@codecademy/gamut@26.2.3-alpha.81eea5.0<br/>@codecademy/styleguide@30.5.2-alpha.81eea5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://604992664f76640b6f95884f--gamut-preview.netlify.app>https://604992664f76640b6f95884f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/604992664f76640b6f95884f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.14-alpha.1b8e8c.0<br/>@codecademy/gamut-labs@11.2.2-alpha.1b8e8c.0<br/>@codecademy/gamut@26.2.3-alpha.1b8e8c.0<br/>@codecademy/styleguide@30.5.2-alpha.1b8e8c.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.14-alpha.8349c0.0<br/>@codecademy/gamut-labs@11.2.2-alpha.8349c0.0<br/>@codecademy/gamut@26.2.3-alpha.8349c0.0<br/>@codecademy/styleguide@30.5.2-alpha.8349c0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://604a3cdcbf3ebe13c0e71cd2--gamut-preview.netlify.app>https://604a3cdcbf3ebe13c0e71cd2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/604a3cdcbf3ebe13c0e71cd2>Deploy Logs</a></p>

**Review by jakemhiller — APPROVED** [review]
lgtm

File: packages/gamut/src/Card/index.tsx [hunk 1]
```diff
@@ -1,8 +1,106 @@
...
+const SHADOWS = {
+  foreground: [1, -1],
+  background: [-2, 2],
+};
+
+const cardVariants = variant({
+  yellow: {
+    backgroundColor: 'yellow',
+    textColor: 'navy',
+    borderColor: 'navy',
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Card/index.tsx:19]: `borderColor` is `'navy'` in all of these. Can it be moved to styles always applied?

**Review by JoshuaKGoldberg — APPROVED** [review]
🚀

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.19-alpha.531202.0<br/>@codecademy/gamut-labs@11.4.2-alpha.531202.0<br/>@codecademy/gamut@26.5.1-alpha.531202.0<br/>@codecademy/styleguide@30.8.1-alpha.531202.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6050ba61ee00f3f104703758--gamut-preview.netlify.app>https://6050ba61ee00f3f104703758--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6050ba61ee00f3f104703758>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.20-alpha.c2e674.0<br/>@codecademy/gamut-labs@11.4.3-alpha.c2e674.0<br/>@codecademy/gamut@26.5.2-alpha.c2e674.0<br/>@codecademy/styleguide@30.8.2-alpha.c2e674.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6050bc4f688cf5d0123fffa0--gamut-preview.netlify.app>https://6050bc4f688cf5d0123fffa0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6050bc4f688cf5d0123fffa0>Deploy Logs</a></p>

**Pull request merged**